.pre-commit-config.yaml

```diff
@@ -0,0 +1,20 @@
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v6.0.0
+  hooks:
+  - id: trailing-whitespace
+  - id: end-of-file-fixer
+  - id: check-yaml
+  - id: check-json
+  - id: mixed-line-ending
+  - id: check-merge-conflict
+  - id: debug-statements
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  # Ruff version.
+  rev: v0.12.10
+  hooks:
+    # Run the formatter.
+    - id: ruff-format
+    # Run the linter.
+    - id: ruff-check
+      args: [ --fix ]
```

Comment:
Fixing is useful for local development, but as part of CI its not since we do want the issues to be reported.

Reply:
I believe that pre-commit flags a check as failed if any files are modified, even if the check returns a zero exit code. So I believe this would work as is, even if all the issues ruff check finds are fixable. Just to be sure, I added the `--exit-non-zero-on-fix` flag to make sure we get a non-zero exit code even if all the issues found are fixable.